<launch>
    <arg name="debug"               default="false"/>
    <arg name="bag_filename"        default="$(env HOME)/bags/campusrover_bag/20200724_dorm2_elevator_enter_1.bag"/>

    <arg name="global_path"       default="/global_path"/>

    <!-- /////////////////////////////////////////////////////////////////////////////////// -->
    <group if="$(arg debug)">

        <param name="use_sim_time" value="true" />
        <node name="playbag" pkg="rosbag" type="play"
                args="--clock $(arg bag_filename)" />

        <param name="robot_description"
            textfile="$(find campusrover_description)/urdf/campusrover.urdf" />

        <node name="robot_state_publisher" pkg="robot_state_publisher"
            type="robot_state_publisher" />


        <!-- rviz -->
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find campusrover_move)/rviz/simulation.rviz" required="true"/>

    </group>

    <node pkg="campusrover_move" type="lu_simulation" name="lu_simulation" output="screen">
        <remap from="global_path"    to="$(arg global_path)" />
        <param name="robot_frame" value="base_link" />
    </node>
</launch>